tests/test_tensor.py

```diff
@@ -46,15 +60,38 @@ def test_common_tensor_backends(backend):
         == [[1, 1, 1], [2, 3, 4], [5, 6, 7]]
     assert list(map(tb.tolist, tb.simple_broadcast([1], [2, 3, 4], [5, 6, 7]))) \
         == [[1, 1, 1], [2, 3, 4], [5, 6, 7]]
-    assert tb.tolist(tb.ones((4, 5))) == [[1.] * 5] * 4
-    assert tb.tolist(tb.zeros((4, 5))) == [[0.] * 5] * 4
-    assert tb.tolist(tb.abs([-1, -2])) == [1, 2]
     with pytest.raises(Exception):
         tb.simple_broadcast([1], [2, 3], [5, 6, 7])
 
+def test_reshape(backend):
+    tb = pyhf.tensorlib
+    assert tb.tolist(tb.reshape(tb.ones((1,2,3)), (-1,))) == [1, 1, 1, 1, 1, 1]
+
+def test_shape(backend):
+    tb = pyhf.tensorlib
+    assert tb.shape(tb.ones((1,2,3,4,5))) == (1,2,3,4,5)
+
+def test_pdf_calculations(backend):
+    tb = pyhf.tensorlib
+    assert tb.tolist(
+        tb.normal_cdf(tb.astensor([0.8]))) == pytest.approx([0.7881446014166034], 1e-07)
     # poisson(lambda=0) is not defined, should return NaN
     assert tb.tolist(pyhf.tensorlib.poisson([0, 0, 1, 1], [0, 1, 0, 1])) == pytest.approx([np.nan, 0.3678794503211975, 0.0, 0.3678794503211975], nan_ok=True)
 
+def test_1D_gather(backend):
+    tb = pyhf.tensorlib
+    assert tb.tolist(tb.gather(tb.astensor([1,2,3,4,5,6]), tb.astensor([4,0,3,2], dtype='int'))) == [5, 1, 4, 3]
```

Comment:
an important case we use extensively in the new code #285 is where the index tensor has some non-trivial structure

e.g.

```
tb.gather(tb.astensor([1,2,3,4,5,6]), tb.astensor([[4,0],[3,2]], dtype='int')) == [[5, 1], [4, 3]]
```
we should add these as well since it's a subtle feature that might be implemented differently in backends